core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -1724,6 +1751,39 @@ object Parser extends ParserInstances {
       null.asInstanceOf[A]
     }
 
+    final def stringIn1[A](radix: RadixNode, all: List[String], state: State): Unit = {
+      val startOffset = state.offset
+      var offset = state.offset
+      var tree = radix
+      var cont = offset < state.str.length
```

Comment:
might be worth a minor optimization to capture state.str.length in a local val to make it clear to the compiler/jit that does not change.